### ISSUE TITLE
F# project does not work with NuGet lockfile

### ISSUE THREAD

**Issue by purkhusid** [dotnet/fsharp#12315]
Please provide a succinct description of the issue.

1. Create a new F# project with the dotnet CLI
2. Add the following `Directory.Packages.props`
```
<Project>
  <PropertyGroup>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
    <RestoreLockedMode>true</RestoreLockedMode>
  </PropertyGroup>
  <ItemGroup>
    <PackageVersion Include="FSharp.Core" Version="5.0.2" Pin="true" />
    <PackageVersion Include="Newtonsoft.Json" Version="10.0.1" Pin="true" />
  </ItemGroup>
</Project>
```
3. Run `dotnet restore --force-evaluate`

**Expected behavior**

The project should be restored and a packages.lock.json file should be generated

**Actual behavior**

Fails with an error
```
error NU1008: Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items: FSharp.Core
```

**Known workarounds**
Add `<DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference>` to the fsproj

**Related information**

Provide any related information (optional):

* Operating system
Fedora Linux 35
* .NET Runtime kind (.NET Core, .NET Framework, Mono)
.NET 5

**Comment by KevinRansom** [comment]
The purpose of ImplicitFSharpCoreReference is to allow the SDK to specify exactly which version of the FSharp.Core nuget package is required.  To provide that information another way, requires it to be disabled, and a specific reference added to the project.

```` <DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference> ``` is  the actual solution.

So I believe this is not a bug, but by design.  Probably the documentation could be enhanced to call this out in the case of lock files.

I hope this helps

Kevin

**Comment by purkhusid** [comment]
Yeah, I think it would be an acceptable solution if the error message prints out the solution to the problem.

**Comment by slang25** [comment]
The problem we have is that as soon as you add a `Directory.Packages.props` file and start using features like centrally managed package versions, it is incompatible with the implicit `FSharp.Core` reference, and the error message doesn't help point you to the fix, which is to follow the instructions in the OP here.

I think the NuGet `NU1008` warning should be aware of this special case, or there be some other metadata flowed through to help identify this scenario.

Ultimately we need something that indicates to the inner NuGet machinery & VS UI that this package is implicitly defined by default, but you *can* change the package version in the VS UI, which should result in disabling the implicit reference and add a local explicit one (i.e. not use `Update="FSharp.Core"`).

That would suggest to me that the best place to address this would be in the VS UI, and to add `IsImplicitlyDefined` as [started here](https://github.com/dotnet/fsharp/pull/6730).

**Comment by vzarytovskii** [comment]
I can't see it anymore in rc2:



![Image](https://github.com/dotnet/fsharp/assets/1260985/be940d29-5ed2-4214-8ca9-fb070ecc85a2)
